new_game.rb

```diff
@@ -0,0 +1,42 @@
+class NewGame
+    def initialize(number_range, no_of_chances = 6)
+        @input = nil
+        @guess = nil
+        @range = number_range
+        @random_number = rand(@range[0]..@range[1])
+        @chances = no_of_chances
+    end
+
+    def start
+        puts "Enter a number between #{@range[0]} and #{@range[1]}, then press enter to submit your guess. You have #{@chances} chances to guess the correct number"
+        get_guess
+        attempts = 0
+        while attempts < ( @chances - 1 )
+            attempts = attempts + 1
+            response = check_guess
+            puts response ? response : break
+            get_guess
+        end
+        puts is_correct ? "Correct! You win" : "Sorry, you ran out of guesses. The number was #{@random_number}"
+    end
+
+    def get_guess
+        @input = gets.chomp
+        @guess = @input.to_i
+    end 
+
+    def check_guess
+        response = nil
```

Comment:
Ruby is also a bit cleve whereby it returns the method result. For example here you don't actually need a `response` instance variable.

Reply:
Cool makes sense - although I want to return nil when there is no response in order to use my ternery operator on line 17 - is there a better way to do this?